File: src/dataset/cpp/kindergarten-garden_exercism-toemm/gemini/GeminiFlash_kindergarten_garden_v2.cpp
#include "kindergarten_garden.h"

#include <array>

namespace kindergarten_garden {

constexpr int FLOWERS_PER_STUDENT = 2;

std::array<Plants, 4> plants(std::string_view flowers, std::string_view student) {
    const int student_index = student[0] - 'A';
    const int flowers_index = FLOWERS_PER_STUDENT * student_index;
    const size_t row_length = flowers.size() / 2;

    return {
        static_cast<Plants>(flowers[flowers_index]),
        static_cast<Plants>(flowers[flowers_index + 1]),
        static_cast<Plants>(flowers[row_length + flowers_index]),
        static_cast<Plants>(flowers[row_length + flowers_index + 1])
    };
}

}  // namespace kindergarten_garden

File: src/dataset/Java/kindergarten-garden_exercism-johannesrave-education/gemini/GeminiFlash_KindergartenGarden_v2.java
import java.util.ArrayList;
import java.util.List;

class KindergartenGarden {
    private final String garden;
    private final int rowLength;

    KindergartenGarden(String garden) {
        String[] rows = garden.split("\n");
        this.garden = String.join("", rows);
        this.rowLength = this.garden.length() / 2;
    }

    List<Plant> getPlantsOfStudent(String student) {
        int studentIndex = student.charAt(0) - 'A';
        int firstPlot = studentIndex * 2;
        int secondPlot = firstPlot + 1;
        int thirdPlot = firstPlot + rowLength;
        int fourthPlot = thirdPlot + 1;

        List<Plant> plants = new ArrayList<>(4);
        plants.add(Plant.getPlant(garden.charAt(firstPlot)));
        plants.add(Plant.getPlant(garden.charAt(secondPlot)));
        plants.add(Plant.getPlant(garden.charAt(thirdPlot)));
        plants.add(Plant.getPlant(garden.charAt(fourthPlot)));

        return plants;
    }
}